File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.cpp
class Solution {
   public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n);
        vector<int> left(n, 1);
        vector<int> right(n, 1);
        int leftP = 1;
        int rightP = 1;
        for (int i = 1; i < n; i++) {
            left[i] = leftP * nums[i - 1];
            leftP = left[i];
        }
        for (int r = n - 2; r >= 0; r--) {
            right[r] = rightP * nums[r + 1];
            rightP = right[r];
        }
        for (int i = 0; i < n; i++) {
            res[i] = left[i] * right[i];
        }
        return res;
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []
        res=[]
        q=collections.deque()
        q.append(root)
        while q:
            nq=[]
            level=[]
            for node in q:
                level.append(node.val)
                if node.left:
                   nq.append(node.left)
                if node.right:
                    nq.append(node.right)
            q=nq
            res.append(level[-1])
        return res 


        
        